src/linux/backend-v4l2.cpp

```diff
@@ -805,6 +805,130 @@ namespace librealsense
             return info;
         }
 
+        bool v4l_uvc_device::is_format_supported_on_node(const std::string& dev_name, std::string v4l_4cc_fmt)
+        {
+            int fd = open(dev_name.c_str(), O_RDWR);
+            if (fd < 0)
+                throw linux_backend_exception("Mipi device format could not be grabbed");
+
+            struct v4l2_fmtdesc fmtdesc;
+            memset(&fmtdesc,0,sizeof(fmtdesc));
+            fmtdesc.type = V4L2_BUF_TYPE_VIDEO_CAPTURE;
+
+            uint32_t format;
+            memcpy(&format, v4l_4cc_fmt.c_str(), sizeof(format));
+
+            while (ioctl(fd,VIDIOC_ENUM_FMT,&fmtdesc) == 0)
```

Comment:
Do you feel OK with the while ?
I think it means that if you cannot read the formats you will loop here forever, maybe we should limit it?

Reply:
same while used in probe_and_commit method - seems ok